### ISSUE TITLE
Unclosed comment crashes JSHint

### ISSUE THREAD

**Issue by fix-fix** [jshint/jshint#2932]
JSHint version: 2.9.2 (I tested it on http://jshint.com/)
Environment: Chrome v51

When there is a open comment block above the property in object, JSHint crashes with error:
```
Uncaught TypeError: Cannot read property 'id' of null
e.nud @ jshint.js:6
C @ jshint.js:5
(anonymous function) @ jshint.js:6
C @ jshint.js:5
fe @ jshint.js:6
pe @ jshint.js:6
gt @ jshint.js:7
self.onmessage @ worker.js:7
```
I tested in on http://jshint.com/ so the message is a bit cryptic, but you can reproduce the bug with the following snippet
```js

({
  /*
  prop: 0
})

```

JSHint should give "Unrecoverable syntax error" error instead of crashing.

**Comment by jugglinmike** [comment]
Thanks for the report! I just submitted a patch for this at gh-2937. Hopefully we'll have the fix out to you soon.

**Comment by Arcanemagus** [comment]
Just so it's a little easier searching for this for anyone in the future, here is the full stack trace from running into this on the CLI:

```text
C:\Temp\github\linter-jshint_GH292\node_modules\jshint\src\jshint.js:5309
        throw err;
        ^

TypeError: Cannot read property 'id' of null
    at Object.x.nud (C:\Temp\github\linter-jshint_GH292\node_modules\jshint\src\jshint.js:3228:26)
    at expression (C:\Temp\github\linter-jshint_GH292\node_modules\jshint\src\jshint.js:884:34)
    at Object.fud (C:\Temp\github\linter-jshint_GH292\node_modules\jshint\src\jshint.js:3666:17)
    at expression (C:\Temp\github\linter-jshint_GH292\node_modules\jshint\src\jshint.js:881:32)
    at statement (C:\Temp\github\linter-jshint_GH292\node_modules\jshint\src\jshint.js:1632:9)
    at statements (C:\Temp\github\linter-jshint_GH292\node_modules\jshint\src\jshint.js:1677:16)
    at itself (C:\Temp\github\linter-jshint_GH292\node_modules\jshint\src\jshint.js:5288:9)
    at lint (C:\Temp\github\linter-jshint_GH292\node_modules\jshint\src\cli.js:495:8)
    at C:\Temp\github\linter-jshint_GH292\node_modules\jshint\src\cli.js:656:7
    at Array.forEach (native)
```

**Comment by jugglinmike** [comment]
Resolved via gh-2937. You can expect the fix in the release that follows 2.9.2.
